dashboard/app/models/script.rb

```diff
@@ -374,6 +383,20 @@ def self.get_from_cache(id_or_name)
     end
   end
 
+  # Given a script family name, return a dummy Script with redirect_to field
+  # pointing toward the latest stable script in that family. For now, we assume:
+  #   1. the 2017 version is the latest stable version
+  #   2. the 2017 version's name is the family_name with the '-2017' suffix
+  # @param family_name [String] The name of the script family to search in.
+  # @return [Script|nil] A dummy script object, not persisted to the database,
+  #   with only the redirect_to field set.
+  def self.get_script_family_redirect(family_name)
+    script_version_name = "#{family_name}-#{2017}"
```

Comment:
I haven't spelled this out in the comments, but I'm entirely ignoring the `family_name` field here because I realized that it doesn't provide enough information for me to find the 2017 version of the course unless I also add a version_year field (or possibly an is_stable field) to each script which as a family_name. I want to do this in the long run, but am proposing punting on this until after the end-of-week deadline.